src/Mutator/Arithmetic/BitwiseNot.php

```diff
@@ -22,11 +22,11 @@
      *
      * @param Node $node
      *
-     * @return mixed
+     * @return \Generator
      */
-    public function mutate(Node $node)
+    public function mutate(Node $node): \Generator
     {
-        return $node->expr;
+        yield $node->expr;
```

Comment:
We're loosing flexibility here. E.g. if it would be allowed to return either simple values, or a generator, not only this upgrade would not require so many changes all around, some future mutators would be easier.

E.g. try refactoring this to use generators without a whole bunch of else/elseif:
```
if ($a) {
    return $k;
}

if ($b) {
    return $c;
}

if ($d) {
    return $k;
}
```
In the same time detecting generators from simple values is super easy:
```
if ($input instanceof Generator) {....
```

Reply:
Not use I got what you mean and what you suggest.
Regarding code example, with generator it would be:

```php
if ($a) {
    yield $k;
    return;
}

if ($b) {
    yield $c;
    return;
}

if ($d) {
    yield $k;
    return;
}
```

No `else`/`elseif` if needed.

And we have such case: https://github.com/infection/infection/pull/450/files#diff-9908f23125ee671a485bd3a39b546da0R32

Reply:
Yeah, that's my point. Instead of a simple and clear `return` one has to use yield-return combo which could make everything very hard to understand because, for example, if you `yield` something twice, even if you return, you got to keep two values, and that's kind of expected for `yield`, but that's not necessarily easy to see from the code.

May I suggest we only use `yield` only where it is actually needed?

Reply:
Oh, and there's a fun part about returning from a generator. [It isn't exactly doing what you can expect.](https://3v4l.org/JLN73)

Reply:
Generators are indeed a bit funny but I would either return a generator yielding the mutations or an array. Having mixed single value + generator/array looks more confusing than anything.

Maybe when you are using a specific mutator and expects only one mutation returning a single value would be a bit counter-intuitive, but I would argue 99% you do not care that it's a single value or multiple ones.

Wether we pick generators or arrays however is how you guys prefers it, given the usage, even if I have a personal preference for generators, I do think arrays will be simpler and less error-prone